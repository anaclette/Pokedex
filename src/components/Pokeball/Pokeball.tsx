import React from 'react';
import {ImageStyle, StyleProp, Image, ImageSourcePropType} from 'react-native';

interface PokeballProps {
  style: StyleProp<ImageStyle>;
  source: ImageSourcePropType | undefined;
}

export const Pokeball = ({style, source}: PokeballProps) => {
  return (
    <Image
      accessibilityRole="image"
      accessibilityLabel="Pokeball image"
      source={source}
      style={style}
    />
  );
};
