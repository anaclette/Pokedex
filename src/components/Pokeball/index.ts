import {Pokeball} from './Pokeball';

export default Pokeball;
