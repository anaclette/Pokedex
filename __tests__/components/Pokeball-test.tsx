import * as React from 'react';
import Pokeball from '../../src/components/Pokeball';
import {render, screen} from '@testing-library/react-native';

describe('Pokeball image renders correctly', () => {
  it('displays the pokeball', () => {
    const component = render(
      <Pokeball style={{width: 20}} source={{uri: 'some source uri'}} />,
    );

    expect(screen.getByLabelText('Pokeball image')).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
