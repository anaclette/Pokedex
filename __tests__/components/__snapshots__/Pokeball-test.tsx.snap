// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokeball image renders correctly displays the pokeball 1`] = `
<Image
  accessibilityLabel="Pokeball image"
  accessibilityRole="image"
  source={
    Object {
      "uri": "some source uri",
    }
  }
  style={
    Object {
      "width": 20,
    }
  }
/>
`;
